compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1826,6 +1832,258 @@ public void visit(BLangWorkerReceive workerReceiveExpr) {
+            }
+        }
+        // If the record literal is of record type and types are validated for the fields, check if there are any
+        // required fields missing.
+        checkMissingReqFieldsForWorkerMultiReceive(((BRecordType) expType),
+                bLangWorkerMultipleReceive.receiveFields, bLangWorkerMultipleReceive.pos);
+
+        if (symTable.semanticError != resultType) {
+            resultType = expType;
+        }
+    }
+
+    private void checkMissingReqFieldsForWorkerMultiReceive(BRecordType type,
+                                                            List<WorkerMultipleReceiveNode.WorkerReceiveFieldNode>
+                                                                    receiveFields, DiagnosticPos pos) {
+
+        for (BField field : type.fields.values()) {
+            // Check if `field` is explicitly assigned a value in the record literal
+            boolean hasField = false;
+            for (WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField : receiveFields) {
+                String receivedFieldIdentifier = receiveField.getWorkerFieldName() != null ?
+                        receiveField.getWorkerFieldName().getValue() : receiveField.getWorkerName().getValue();
+                if (field.name.value.equals(receivedFieldIdentifier)) {
+                    hasField = true;
+                    break;
+                }
+            }
+
+            // If a required field is missing, it's a compile error
+            if (!hasField && Symbols.isFlagOn(field.symbol.flags, Flags.REQUIRED)) {
+                dlog.error(pos, DiagnosticCode.MISSING_REQUIRED_RECORD_FIELD, field.name);
+            }
+        }
+    }
+
+    private void checkWorkerMultipleReceiveField(WorkerMultipleReceiveNode.WorkerReceiveFieldNode receiveField,
+                                                 BType type) {
+
+        BLangExpression expression;
+        if (receiveField.getKeyExpression() != null) {
+            BSymbol symbol = symResolver.lookupSymbolInMainSpace(env,
+                    names.fromIdNode(((BLangSimpleVarRef) receiveField.getKeyExpression()).variableName));
+            ((BLangSimpleVarRef) receiveField.getKeyExpression()).type = symbol.type;
+            expression = (BLangExpression) receiveField.getKeyExpression();
+        } else {
+            expression = receiveField.getSendExpression();
+        }
+
+        if (receiveField.getWorkerFieldName() != null) {
+            // Type check with worker field name
+            BSymbol fieldSymbol = symResolver.lookupSymbolInMainSpace(
+                    env,
+                    names.fromIdNode((
+                                    (BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField).workerFieldName
+                                    ));
+            ((BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField).workerFieldName.type
```

Comment:
shall we extract `((BLangWorkerMultipleReceive.BLangWorkerReceiveField) receiveField)` to a variable here.

Reply:
Fix the formatting as well. Saw multiple other places as well, with formatting issues

Reply:
Fixed, @pubudu91 fixed the formatting issue in the other places as well